docs/reference/cli.md

```diff
@@ -216,6 +216,13 @@ pixi upgrade --json
 pixi upgrade --dry-run
 ```
 
+!!! note
+    The `pixi upgrade` command will only update `version`s, except when you specify the exact package name (`pixi upgrade numpy`).
+
+    Then it will also update the following fields: `version`, `build`, `build_number`, `md5`, `sha265`.
+
+    Except a `build` field with a wildcard `*` will always be reused in the upgrade, the same goes for `channel`, `file_name`, `url`, and `subdir`.
```

Comment:
That is a long decision tree :)

How about this:

```suggestion
    Then it will remove all fields, apart from:
    - `build` field containing a wildcard `*`
    - `channel`
    - `file_name`
    - `url`
    - `subdir`.
```